3-UsingAPIs/Week1/assignment/ex1-johnWho.js

```diff
@@ -7,22 +7,25 @@ Rewrite this function, but replace the callback syntax with the Promise syntax:
 - If the Promise `rejects`, pass an error as the argument to reject with: "You 
   didn't pass in a first name!"
 ------------------------------------------------------------------------------*/
-// TODO see above
-export const getAnonName = (firstName, callback) => {
-  setTimeout(() => {
-    if (!firstName) {
-      callback(new Error("You didn't pass in a first name!"));
-      return;
-    }
+export const getAnonName = (firstName) => {
 
-    const fullName = `${firstName} Doe`;
+  return new Promise((resolve, reject) => {
+      if (!firstName) {
+        reject(new Error("You didn't pass in a first name!"))
+      } else {
+        const fullName = `${firstName} Doe`;
+        resolve(fullName)
+      };
```

Comment:
Excellent! 

1. You can also "exit early" when rejecting the promise. The idea is to check for errors at the start of the function body and return "early" in case of errors. This eliminates the need for an `else` clause and also requires one less indentation level on the remainder of the code. For a short function like the one here it does not make much of a difference but when you function becomes larger it will improve readability.

2. Your VSCode Prettier setup is not working correctly as the code is not automatically formatted according to the Prettier rules. Send me a DM on Slack so we can fix that for you.

```suggestion
    if (!firstName) {
      reject(new Error("You didn't pass in a first name!"));
      return;
    }

    const fullName = `${firstName} Doe`;
    resolve(fullName);
  });
```